tsdb/index/index.go

```diff
@@ -1268,7 +1270,10 @@ func newReader(b ByteSlice, c io.Closer) (*Reader, error) {
 		r.nameSymbols[off] = k
 	}
 
-	r.dec = &Decoder{LookupSymbol: r.lookupSymbol}
+	if postingsDecoder == nil {
+		postingsDecoder = DecodePostingsRaw
```

Comment:
There are three places where `nil` is reinterpreted as a default.  Do we need three?

Reply:
Removed this one and `newReader` always requires to pass an not nill function

Reply:
Thanks; we're down to two now and neither is `newReader`, so can you comment on those two please?